balena/models/device.py

```diff
@@ -1443,7 +1443,7 @@ def start_os_update(self, uuid, target_os_version):
         self.__check_os_update_target(device, target_os_version)
 
         all_versions = self.device_os.get_supported_versions(device['is_of__device_type'][0]['slug'])['versions']
-        if not [v for v in all_versions if semver.compare(target_os_version, v) == 0]:
+        if not [v for v in all_versions if target_os_version == v['stripped_version']]:
```

Comment:
Afaict this would prevent users to use the full version `2.29.2+rev1.prod`, which sounds like a breaking change.
Afaik the HUP script does expect the fully qualified version, including the `.prod`/`.dev` if the version is a non-unified one.
How about v ?
```suggestion
        if not [v for v in all_versions if target_os_version == v['raw_version']]:
```

Reply:
@thgreasi, I used `stripped_version` here since we do not include the `.prod`/`.dev` build metadata part in the balenaOS version list in the dashboard but you are right, `raw_version` will cover the fully qualified version case. Changed to `raw_version`